src/mercury-rpc/mercury/rpc/mongo.py

```diff
@@ -0,0 +1,75 @@
+from mercury.common import mongo as common_mongo
+from mercury.common.asyncio import mongo as asyncio_mongo
+
+__rpc_collections = None
+
+
+class RPCCollectionFactory(object):
+    """ The purpose of this class is to provide the multiple RPC layer services
+    a single place to configure and instantiate database connections. """
+
+    def __init__(self, servers, database, jobs_collection, tasks_collection,
+                 replica_name=None, use_asyncio=False):
+        """ Constructor
+
+        :param servers: MongoDB servers to connect to connect to
+        :type servers: list
+        :param database: The database to use for both collections
+        :param jobs_collection: The name of the jobs collection
+        :param tasks_collection: The name of the tasks collection
+        :param replica_name: The name of the replica set to use, if any
+        :param use_asyncio: Use asyncio for the connection object or not
+        """
+        self.servers = servers
+        self.database = database
+        self.jobs_collection_name = jobs_collection
+        self.tasks_collection_name = tasks_collection
+        self.replica_name = replica_name
+        self.use_asyncio = use_asyncio
+
+        # Caches
+        self.__connection = None
+        self.__jobs = None
+        self.__tasks = None
+
+        # Store instance
+        global __rpc_collections
+        __rpc_collections = self
```

Comment:
This will blow up something at some point.

Reply:
?